### PR TITLE
feat(suite): use current backend url if connected/defined

### LINKED ISSUE #4044: use current connected backend url for explorer links
`transactionUtils.getBlockExplorerUrl` should use current connected backend url (from the reducer) if connected/defined and if not should fallback for default (current sollution)

### PR DESCRIPTION
**Pull request by dimasivan**
fix #4044

### CODE REVIEW

File: packages/suite/src/components/suite/modals/TransactionDetail/components/ChainedTxs/index.tsx [hunk 1]
```diff
@@ -16,12 +16,16 @@ const StyledTrezorLink = styled(TrezorLink)`
...
+    explorerUrl?: string;
 }
 
-const ChainedTxs = ({ txs }: Props) => (
+const ChainedTxs = ({ txs, explorerUrl }: Props) => (
     <Wrapper>
         {txs.map(tx => (
-            <StyledTrezorLink href={getBlockExplorerUrl(tx)} variant="nostyle">
+            <StyledTrezorLink
+                href={explorerUrl ? `${explorerUrl}/tx/${tx.txid}` : getBlockExplorerUrl(tx)}
```
Comment (szymonlesisz) [inline packages/suite/src/components/suite/modals/TransactionDetail/components/ChainedTxs/index.tsx]: you can do this check only once: in `TransactionDetail/index` and pass it here and into `BasicDetails` as required string

File: packages/suite/src/components/suite/modals/TransactionDetail/index.tsx [hunk 2]
```diff
@@ -55,10 +55,12 @@ type Props = {
 
 const TransactionDetail = (props: Props) => {
     const { tx } = props;
+
     const { blockchain, transactions } = useSelector(state => ({
         blockchain: state.wallet.blockchain[tx.symbol],
         transactions: state.wallet.transactions.transactions,
     }));
+
```
Comment (szymonlesisz) [inline packages/suite/src/components/suite/modals/TransactionDetail/index.tsx]: this is where you should do:
`blockchain.url ? `${blockchain.url}/tx/${tx.txid}` : getBlockExplorerUrl(tx)`

and pass it lower

Comment (dimasivan) [reply packages/suite/src/components/suite/modals/TransactionDetail/index.tsx]: I can do it for `BasicDetail` component, but `ChainedTxs` receives multiple txs in array and then mapping it in render

File: packages/suite/src/components/suite/modals/TransactionDetail/components/ChainedTxs/index.tsx (hunk 1 above)
Comment (dimasivan) [reply packages/suite/src/components/suite/modals/TransactionDetail/components/ChainedTxs/index.tsx]: fixed in 2ea85cd

File: packages/suite/src/components/suite/modals/TransactionDetail/index.tsx (hunk 2 above)
Comment (dimasivan) [reply packages/suite/src/components/suite/modals/TransactionDetail/index.tsx]: 2ea85cd

**Review by szymonlesisz — APPROVED** [review]

**Pull request merged**